### PR TITLE
Nü passcode screen: V

### LINKED ISSUE #761: Nü Passcode
### Guidelines
- New code should always be Swift
- Autolayout **is always preferred**
- Please drop outdated components when appropriate! 🔥🔥🔥

### Details
- [ ] Encapsulate all of the Pincode support (currently spread in the AppDelegate)
- [ ] Encapsulate our FaceId + TouchID support
- [ ] Unit Test all the things
- [ ] Implement the new Passcode UI
- [ ] Please keep the Attempts Limit logic

### PR DESCRIPTION
**Pull request by eshurakov**
### Fix
In this PR we're adding unit tests for controllers that provide logic for setup / remove / verify passcode screens.

@jleandroperez sorry, PR went a bit over 500 LOC 🙇 

Related issue: #761

### Test

- [ ] Unit Tests are green

### Review
Only one developer is required to review these changes, but anyone can perform the review.

### CODE REVIEW

File: Simplenote/ApplicationStateProvider.swift [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Foundation
```
Comment (jleandroperez) [inline Simplenote/ApplicationStateProvider.swift]: Nipticky: Maybe move this to `Classes` ?

File: Simplenote/UIApplication+Simplenote.swift [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import UIKit
```
Comment (jleandroperez) [inline Simplenote/UIApplication+Simplenote.swift]: Nipticky: Move this one to classes? 

Ooor maybe just add this to `ApplicationStateProvider.swift` itself?

File: Simplenote/BiometricAuthentication.swift [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import Foundation
```
Comment (jleandroperez) [inline Simplenote/BiometricAuthentication.swift]: Same note as above: Maybe relocate the file to `Classes` ?

File: Simplenote/Classes/PinLock/PinLockRemoveController.swift [hunk 4]
```diff
@@ -16,23 +16,26 @@ final class PinLockRemoveController: PinLockBaseController, PinLockController {
     }
 
     private var attempts: Int = 1
+    private let pinLockManager: SPPinLockManager
     private weak var delegate: PinLockRemoveControllerDelegate?
 
-    init(delegate: PinLockRemoveControllerDelegate) {
+    init(pinLockManager: SPPinLockManager = SPPinLockManager.shared,
```
Comment (jleandroperez) [inline Simplenote/Classes/PinLock/PinLockRemoveController.swift]:
```suggestion
    init(pinLockManager: SPPinLockManager = .shared,
```

Comment (jleandroperez) [reply Simplenote/Classes/PinLock/PinLockRemoveController.swift]: Isn't Swift awesome? 😄

**Review by jleandroperez — APPROVED** [review]
Shiny!! nice work!!

:shipit:

File: Simplenote/Classes/PinLock/PinLockRemoveController.swift (hunk 4 above)
Comment (eshurakov) [reply Simplenote/Classes/PinLock/PinLockRemoveController.swift]: Yes, it is! :-)

File: Simplenote/BiometricAuthentication.swift (hunk 3 above)
Comment (eshurakov) [reply Simplenote/BiometricAuthentication.swift]: Thank you, done!

File: Simplenote/UIApplication+Simplenote.swift (hunk 2 above)
Comment (eshurakov) [reply Simplenote/UIApplication+Simplenote.swift]: Thank you, moved to classes, done!

File: Simplenote/ApplicationStateProvider.swift (hunk 1 above)
Comment (eshurakov) [reply Simplenote/ApplicationStateProvider.swift]: Thank you, done!

**Comment by eshurakov** [comment]
Thank you @jleandroperez!

**Comment by pachlava** [comment]
[UI Testing] Verified on iPhone XR | iOS 14.3 | SN 4.30

**Pull request merged**